test/unit/lib/plugins/aws/package/compile/events/apiGateway/lib/restApi.test.js

```diff
@@ -157,4 +162,45 @@ describe('#compileRestApi()', () => {
     awsCompileApigEvents.serverless.service.provider.vpcEndpointIds = ['id1'];
     expect(() => awsCompileApigEvents.compileRestApi()).to.throw(Error);
   });
+
+  describe('Basic configuration', () => {
```

Comment:
Let's put the `runServerless`-based tests in a separate `top-level` describe with the path to this test file, similarily to e.g. https://github.com/serverless/serverless/blob/078ec59058e1c37bd81388c7e81087b48fc2ba24/test/unit/lib/plugins/aws/package/compile/events/httpApi.test.js#L10

Additionally, let's avoid the nested `describes` and go with just `it` for separate cases, e.g. 
`it('should not disable the default endpoint by default')` and `it('should support provider.apiGateway.disableDefaultEndpoint')` with `runServerles` being run as a part of specific `it` test case. We use the approach with running `runServerless` in `before` to avoid mulptile `runServerless` run when it's not needed, but here we need separate `runServerless` runs for each case.